src/neo/SmartContract/Native/Tokens/Nep11Token.cs

```diff
@@ -0,0 +1,278 @@
+#pragma warning disable IDE0060
+
+using Neo.Cryptography;
+using Neo.IO;
+using Neo.IO.Json;
+using Neo.Ledger;
+using Neo.Persistence;
+using Neo.SmartContract.Manifest;
+using Neo.VM;
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using Array = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract.Native.Tokens
+{
+    public abstract class Nep11Token<TToken, TAccount> : NativeContract
+        where TToken : Nep11TokenState, new()
+        where TAccount : AccountState, new()
+    {
+        public override string[] SupportedStandards { get; } = { "NEP-10", "NEP-11" };
+        public abstract string Symbol { get; }
+        public abstract byte Decimals { get; }
```

Comment:
Do we really need `decimals` greater than `0` in native contracts?

Reply:
We need, because it is only inseparable for NFT assets such as domain names, but some other NFTs that can be split.

Reply:
I prefer we can have two Nep11, Nep11a for `Decimals = 0`, and `Nep11b` for `Decimals > 0`.

Reply:
Decimals could be optional and take 0 as default

Reply:
I think the existing design is enough

Reply:
> We need, because it is only inseparable for NFT assets such as domain names, but some other NFTs that can be split.

What other NFTs? I think it is for native contracts only.

Reply:
According to the NEP-11 specification, if `decimals () == 0`, you must implement:

```c#
public bool transfer(byte[] to, byte[] tokenid)
```

And if `decimals () > 0`, you must implement:

```c#
public bool transfer(byte[] from, byte[] to, BigInteger amount, byte[] tokenid)
```

Currently you only implemented the second override. But I think we only need the first override for NNS.

Reply:
It's ok for me to set `decimals = 0`, we only use it for NNS in native contract now.

Reply:
But if we set `decimals = 0`,  could we use the following interfaces, as it'll be more clearly. If that, we need to update nep11.

```cs
public bool transfer(byte[] to, byte[] tokenid)
public byte[] ownerOf(byte[] tokenid)
public byte[] tokensOf(byte[] owner)
```

Reply:
For `transfer`, yes. For `ownerOf`, no.

Reply:
Done.

Reply:
The implementation of `ownerOf` is wrong.